src/theory/arith/arith_utilities.cpp

```diff
@@ -0,0 +1,196 @@
+/*********************                                                        */
+/*! \file arith_utilities.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of common functions for dealing with nodes.
+ **/
+
+#include "arith_utilities.h"
+
+using namespace CVC4::kind;
+
+namespace CVC4 {
+namespace theory {
+namespace arith {
+
+Kind joinKinds(Kind k1, Kind k2)
+{
+  if (k2 < k1)
+  {
+    return joinKinds(k2, k1);
+  }
+  else if (k1 == k2)
+  {
+    return k1;
+  }
+  Assert(isRelationOperator(k1));
+  Assert(isRelationOperator(k2));
+  if (k1 == EQUAL)
+  {
+    if (k2 == LEQ || k2 == GEQ)
+    {
+      return k1;
+    }
+  }
+  else if (k1 == LT)
+  {
+    if (k2 == LEQ)
+    {
+      return k1;
+    }
+  }
+  else if (k1 == LEQ)
+  {
+    if (k2 == GEQ)
+    {
+      return EQUAL;
+    }
+  }
+  else if (k1 == GT)
+  {
+    if (k2 == GEQ)
+    {
+      return k1;
+    }
+  }
+  return UNDEFINED_KIND;
+}
+
+Kind transKinds(Kind k1, Kind k2)
+{
+  if (k2 < k1)
+  {
+    return transKinds(k2, k1);
+  }
+  else if (k1 == k2)
+  {
+    return k1;
+  }
+  Assert(isRelationOperator(k1));
+  Assert(isRelationOperator(k2));
+  if (k1 == EQUAL)
+  {
+    return k2;
+  }
+  else if (k1 == LT)
+  {
+    if (k2 == LEQ)
+    {
+      return k1;
```

Comment:
the behavior seems similar to joinKinds. I wonder if that is intentional

Reply:
The two are similar but trans applies to 3 terms whereas join applies to 2, for example `a = b ^ b >= c` imples` a >= c` whereas `a=b ^ a>=b` implies `a=b`.

Reply:
I see